### PR TITLE
fix: Write poetryOptions to pyproject.toml

### PR DESCRIPTION
**Pull request by lmarsden**
* The poetryOptions object in the PythonProject was previously ignored when synthesizing pyproject.toml.
* poetryOptions declarations are now written to pyproject.toml.
* This allows specification of e.g. private PyPI registries via `[tool.poetry.source]` entries:

```
poetryOptions: {
    source: [
      {
        name: "myprivateregistry",
        url: "https://pkgs.dev.azure.com/myorg/_packaging/myprivateregistry/pypi/simple/",
        secondary: true,
      },
    ]
  },
```

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by Chriscbr — CHANGES_REQUESTED** [review]
Can you add some tests to `python-project.test.ts` or `poetry.test.ts` that illustrate the change?

**Comment by lmarsden** [comment]
> Can you add some tests to `python-project.test.ts` or `poetry.test.ts` that illustrate the change?

Have now updated the code in `poetry.test.ts` to test for correct behaviour of the `poetry` options that were previously missing. `PoetryPyrojectOptions` (via `PoetryPyrojectOptionsWithoutDeps`) class now updated to support all `poetry` options documented at  https://python-poetry.org/docs/pyproject

File: src/__tests__/python/poetry.test.ts [hunk 1]
```diff
@@ -14,15 +14,73 @@ test('poetry enabled', () => {
...
+  expect(snapshot['pyproject.toml']).toContain('[[tool.poetry.source]]');
+  expect(snapshot['pyproject.toml']).toContain('url = "https://pypi.org/simple/"');
+  expect(snapshot['pyproject.toml']).toContain('[tool.poetry.scripts]');
+  expect(snapshot['pyproject.toml']).toContain('test-python-cli = "test-python-project.cli:cli"');
+  expect(snapshot['pyproject.toml']).toContain('[tool.poetry.extras]');
+  expect(snapshot['pyproject.toml']).toContain('cli = [ "package1", "package2" ]');
+  expect(snapshot['pyproject.toml']).toContain('[tool.poetry.plugins."blogtool.parsers"]');
+  expect(snapshot['pyproject.toml']).toContain('".rst" = "some_module:SomeClass"');
+  expect(snapshot['pyproject.toml']).toContain('[tool.poetry.urls]');
+  expect(snapshot['pyproject.toml']).toContain('"bug tracker" = "https://github.com/test-python-project/issues"');
```
Comment (Chriscbr) [inline src/__tests__/python/poetry.test.ts:83]: This is getting pretty long and detailed - let's replace all of this (in the new test case) with

```js
expect(snapshot['pyproject.toml']).toMatchSnapshot();
```

**Review by Chriscbr — CHANGES_REQUESTED** [review]
Looks great @lmarsden, just a few small comments.

File: src/__tests__/python/poetry.test.ts (hunk 1 above)
Comment (lmarsden) [reply src/__tests__/python/poetry.test.ts]: Thanks for the suggestion. Snapshot added and tests updated.

File: src/__tests__/python/poetry.test.ts [hunk 2]
```diff
@@ -14,15 +14,73 @@ test('poetry enabled', () => {
     classifiers: [
       'Development Status :: 4 - Beta',
     ],
+    poetryOptions: {
```
[the remark this answers is not in the document]
Comment (lmarsden) [reply src/__tests__/python/poetry.test.ts:44]: Done. Also see comment below.

**Review by Chriscbr — APPROVED** [review]

**Comment by lmarsden** [comment]
Looks like the build is failing at the `Commit and push changes (if changed)` step in `build.yml` as it's attempting to push some changes to my fork at https://github.com/lmarsden/projen

Are there any changes I need to make?

**Comment by Chriscbr** [comment]
Looks like some kind of transient snapshot error because of some issues with our integration tests. Usually re-running `yarn build` fixes this

**Pull request merged**